tests/unit/_internal/test_configuration.py

```diff
@@ -66,3 +66,26 @@ def test_bentoml_configuration_runner_override():
     assert test_runner_batching["batching"]["max_batch_size"] == 10
     assert test_runner_batching["logging"]["access"]["enabled"] is True
     assert test_runner_batching["resources"]["cpu"] == 4  # should use global
+
+
+RUNNER_TIMEOUTS = """\
+runners:
+    timeout: 50
+    test_runner_1:
+        timeout: 100
+    test_runner_2:
+        resources: system
+"""
+
+
+def test_runner_timeouts():
+    tmpfile = NamedTemporaryFile(mode="w+", delete=False)
+    tmpfile.write(RUNNER_TIMEOUTS)
+    tmpfile.flush()
+    tmpfile.close()
```

Comment:
hmm should this be a context manager instead?

Reply:
maybe, I'm not sure. The only issue was I wanted to manually close() the `tmpfile` because in windows this will fail to open the file created by NamedTemporary file if it is still open with some other process which is a strange windows issue so in order to avoid that I'm manually handling it.

Reply:
It seems that we are not deleting the temp file after the test

Reply:
thats the case right now. But opening it inside a ctx manager would ensure correct gc.

Another thing you can do is to use `unittest.TestCase`

Reply:
But I think `tearDown` just gc everything, so it doesn't make a lot of difference.

Reply:
yeah, we are not deleting it because of the same issue on windows. Originally once NamedTemporaryFile is closed it will automatically get deleted but we can't do that because In windows the tempfile has to be closed before BentoMLConfiguration can read from it.

but I thought it's okay since it's only in temp and will get cleanup anyways?

Reply:
According to the doc of tempfile, we need to take the responsibility to clean the temp files.

Reply:
I think it's fine since this is only for tests and is created in its own temp directory in pytest